src/tests/test_engine.rs

```diff
@@ -379,3 +379,27 @@ fn assignment_to_in_var_no_panic() -> TestResult {
 fn assignment_to_env_no_panic() -> TestResult {
     fail_test(r#"$env = 3"#, "cannot_replace_env")
 }
+
+#[test]
+fn short_flags() -> TestResult {
+    run_test(
+        r#"def foobar [-a: int, -b: string, -c: string] { echo $'($a) ($c) ($b)' }; foobar -b "balh balh" -a 1543  -c "FALSE123""#,
+        "1543 FALSE123 balh balh",
+    )
+}
+
+#[test]
+fn short_flags_1() -> TestResult {
+    run_test(
+        r#"def foobar [--a: string, --b: string, --s: int] { if ( $s == 0 ) { echo $'($b)($a)' }}; foobar --a test --b case --s 0  "#,
+        "casetest",
+    )
+}
+
+#[test]
+fn short_flags_2() -> TestResult {
+    run_test(
+        r#"def foobar [--a: int, -b: string, --c: int] { $a + $c };foobar -b "balh balh" --a 10  --c 1 "#,
+        "11",
+    )
+}
```

Comment:
i've `nushell` installed on d339902dc and i already have the right results for `short_flags_1` and `short_flags_2`, so i'm not sure these two tests capture the fix here :open_mouth: 

on the other hand, the `short_flags` test does show the right behaviour with your changes :yum:

Reply:
Thanks for review .
you mean last two tests works with your commit , without this PR ?

Reply:
yup, exactly :yum: 

either from the tip of your PR branch, 52fea48b0c17308bb1f5b09806830b997e698da9, or its base, d42c2b2dbc941d63c5a05404a0c22c4a7cd9f965, running the 
```nushell
cargo run --release -- -c "def foobar [--a: string, --b: string, --s: int] { if ( $s == 0 ) { echo $'($b)($a)' }}; foobar --a test --b case --s 0  "
cargo run --release -- -c 'def foobar [--a: int, -b: string, --c: int] { $a + $c };foobar -b "balh balh" --a 10  --c 1 '
```
always gives me
```
casetest
11
```

so i'm wondering if we should find more tests that fail before the PR and pass with your changes, just as the `short_flags` test does :relieved:

Reply:
There is something , when I used `--` for args on your commit it works , when I use `-` , it doesn't . Like the tests .
So as I can see the first argument is long flag which is considered to evaluate before. But not for short flags , `-` .

Reply:
ooooh, so should be `foobar -a test -b case -s 0` instead of `foobar --a test --b case --s 0`?

Reply:
Exactly

Reply:
that would make sense :ok_hand: 
let's see what the CI thinks of this :smirk:

Reply:
ohhh yes, with the short flags, i get `00` and `2`, thanks for the catch and the fix :relieved: [